dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -935,41 +782,49 @@ protected Contentlet find(final String inode) throws ElasticsearchException, Dot
-                " inode.inode<>vi.workingInode and "+
-                " mod_user <> 'system' and inode.identifier = '" + identifier.getInode() + "'" +
-                " and type='contentlet' order by mod_date desc");
-        List<com.dotmarketing.portlets.contentlet.business.Contentlet> fatties = hu.list();
-        if(fatties == null)
-            return cons;
+	protected List<Contentlet> findAllUserVersions(final Identifier identifier) throws DotDataException, DotStateException, DotSecurityException {
+        if(!InodeUtils.isSet(identifier.getId()))
+            return Collections.emptyList();
+
+        final DotConnect dotConnect = new DotConnect();
+        final StringBuilder query = new StringBuilder();
+        query.append("select inode, owner from inode, contentlet_version_info vi ")
+                .append("where vi.identifier=inode.identifier and ")
+                .append("inode.inode<>vi.workingInode and ")
+                .append("mod_user <> 'system' and inode.identifier = ? ")
+                .append("and type='contentlet' order by mod_date desc");
+
+        dotConnect.setSQL(query.toString());
+        dotConnect.addParam(identifier.getId());
+
+        List results = dotConnect.loadObjectResults();
+
+        if(results == null)
+            return Collections.emptyList();
         else{
-            for (com.dotmarketing.portlets.contentlet.business.Contentlet fatty : fatties) {
-                Contentlet content = convertFatContentletToContentlet(fatty);
-                contentletCache.add(String.valueOf(content.getInode()), content);
-                cons.add(content);
-            }
+            return TransformerLocator.createContentletTransformer(results).asList();
         }
-        return cons;
 	}
 
     @Override
-    protected List<Contentlet> findAllVersions(Identifier identifier) throws DotDataException, DotStateException, DotSecurityException {
+    protected List<Contentlet> findAllVersions(final Identifier identifier) throws DotDataException, DotStateException, DotSecurityException {
         return findAllVersions(identifier, true);
     }
 
+    @Override
+    protected List<Contentlet> findAllVersions(final Identifier identifier, final boolean bringOldVersions) throws DotDataException, DotStateException, DotSecurityException {
+        return findAllVersions(identifier, bringOldVersions, Optional.empty());
+    }
+
 	@Override
-	protected List<Contentlet> findAllVersions(Identifier identifier, boolean bringOldVersions) throws DotDataException, DotStateException, DotSecurityException {
-	    if(!InodeUtils.isSet(identifier.getInode()))
-            return new ArrayList<Contentlet>();
+    public List<Contentlet> findAllVersions(final Identifier identifier,
+            final boolean bringOldVersions, final Optional<Integer> maxResults)
+            throws DotDataException, DotStateException, DotSecurityException {
+	    if(!InodeUtils.isSet(identifier.getId()))
```

Comment:
Use curly brackets and returns Collections.emptyList() instead if you do not need to modify the collection on upper layers

Reply:
Done